bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/Cardio2eBinding.java

```diff
@@ -0,0 +1,944 @@
+				            		} else {
+				            			 //Will ask Cardio 2é for current date and time (GET transaction). When Cardio answers, will send new value (SET) if offset is less than datetimeMaxOffset.
+				            			pendingDateTimeUpdate.setDateTime((DateTimeType) newDateTimeState);
+				            			logger.debug("Executing date and time update task. Now will ask Cardio for current date and time...");
+				            			getCardioDateTime();
+				            		}
+			            		} else {
+			            			logger.debug("Date and time update ignored because time is not betwen " + allowedDatetimeUpdateHour + ":00 and " + allowedDatetimeUpdateHour + ":59");
+			            		}
+			            	} else {
+			            		logger.debug("No state sent to Cardio 2è because item is not configured as 'reverse mode'");
+			            	}
+			            }
+					} catch (Exception e) {
+						logger.error("Error processing state. " + e);
+					}
+				}
+	        } else {
+	        	logger.error("No state will sent to Cardio 2è because no provider is available");
+	        }
+		} else {
+			logger.error("Can not send state to Cardio 2è because we are not logged in");
+		}
+	}
+
+	private void getCardioDateTime() {
+		if ((datetimeMaxOffset < 0) || ((System.currentTimeMillis() - lastDateTimeRequestTimestamp) >= DATE_TIME_GET_MIN_DELAY)) {
+			lastDateTimeRequestTimestamp = System.currentTimeMillis();
+			try {
+            	logger.debug("Date and time update in course...");
+				com.sendTransaction(new Cardio2eDateTimeTransaction(Cardio2eTransactionTypes.GET));	            		
+	        	logger.debug("Added GET date and time request to Cardio 2é RS-232 send queue");
+			} catch (Exception e) {
+				logger.error("Failed to send GET date and time request." + e);
+			}
+		}
+	}
+	
+	private class LiveDateTime {
+		private DateTimeType dateTime = null;
+		private long timestamp; 
+		public LiveDateTime(){
+		}
+		public void setDateTime(DateTimeType dateTime) {
+			synchronized (this) {
+				timestamp = System.currentTimeMillis();
+				this.dateTime = dateTime;
+			}
+		}
+
+		public DateTimeType getDateTime() {
+			synchronized (this) {
+				return this.dateTime;
+			}
+		}
+
+		public DateTimeType getUpdatedDateTime() {
+			synchronized (this) {
+				DateTimeType dateTimeUpdate = null;
+				if (this.dateTime != null) {
+					dateTimeUpdate = new DateTimeType();
+					dateTimeUpdate.getCalendar().setTimeInMillis(this.dateTime.getCalendar().getTimeInMillis() + (System.currentTimeMillis() - timestamp));
+				}
+				return dateTimeUpdate;
+			}
+		}
+	}
+	
+	private class ReceivedDataListener implements Cardio2eComEventListener {
+		public ReceivedDataListener(){
+		}
+		public void receivedData(Cardio2eReceivedDataEvent e) {
+			decoder.decodeReceivedCardio2eStream(e.getReceivedData());
+		}
+		public void isConnected(Cardio2eConnectionEvent e) {
+			logger.debug("Cardio is " + (e.getIsConnected() ? "CONNECTED" : "DISCONNECTED") + ".");
```

Comment:
Usage of ? : is confusing in logging statements. This problem also occurs many times.